src/plugins/intel_npu/src/plugin/npuw/weights_bank.cpp

```diff
@@ -117,22 +119,44 @@ void Bank::evaluate_and_allocate() {
                 return;
             }
 
+            // Need to allocate
+            // Note: secured by the lock above
+            uids_to_allocated.insert({device_bank.registered_tensors.at(lt), ov::Tensor()});
+        });
+
+        // Allocate memory sequentially - in order of UID
+        for (const auto& elem : uids_to_allocated) {
+            auto uid = elem.first;
+            const auto& transformed_tensor = device_bank.storage[uid].tensor;
```

Comment:
testing on CPU only is ok but you shouldn't forget about the primary target. This code would clearly not work there?